sdk/metrics/src/main/scala/org/typelevel/otel4s/sdk/metrics/exemplar/TraceContextLookup.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2024 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk.metrics.exemplar
+
+import org.typelevel.otel4s.sdk.context.Context
+import org.typelevel.otel4s.sdk.metrics.data.ExemplarData.TraceContext
+
+/** Provides a way to extract [[TraceContext]] from the [[Context]].
+  */
+trait TraceContextLookup {
```

Comment:
Here is an example of the 'real' implementation: https://github.com/iRevive/otel4s/blob/e5c1d041f4a7b025ee264c2358996f9e3a6031b1/sdk/all/src/main/scala/org/typelevel/otel4s/sdk/OpenTelemetrySdk.scala#L381-L394

@NthPortal I still keep in mind your idea to move `TraceContext` to the `core-common`. Perhaps we can even share the `Context.Key[TraceContext]` that way. Once we get closer to the final implementation, I will revisit it.